robolectric/src/test/java/org/robolectric/shadows/ShadowCanvasTest.java

```diff
@@ -529,4 +530,36 @@ public void restoreToCount() {
 
     assertThat(canvas.getSaveCount()).isEqualTo(save1);
   }
+
+  @Test
+  public void drawRect_toCanvas() {
+    Bitmap bitmap = Bitmap.createBitmap(10, 10, Bitmap.Config.ARGB_8888);
+    bitmap.eraseColor(Color.BLACK);
+    assertThat(bitmap.getPixel(0, 0)).isEqualTo(Color.BLACK);
+
+    Canvas canvas = new Canvas(bitmap);
+    Paint paint = new Paint();
+    paint.setStyle(Paint.Style.FILL);
+    paint.setAntiAlias(true);
+    paint.setColor(Color.WHITE);
+    canvas.drawRect(0, 0, 10, 10, paint);
+    assertThat(bitmap.getPixel(0, 0)).isEqualTo(Color.WHITE);
+  }
+
+  @Test
+  public void draw_Rect() throws Exception {
+    Canvas canvas = new Canvas();
+    Paint paint = new Paint();
+    paint.setColor(Color.WHITE);
+    RectF rect = new RectF(2f, 4f, 10f, 10f);
+    canvas.drawRect(2f, 4f, 10f, 10f, paint);
+    ShadowCanvas shadowCanvas = shadowOf(canvas);
+
+    assertThat(shadowCanvas.getDrawnRect(0).left).isEqualTo(2f);
```

Comment:
What about extract the result of `shadowCanvas.getDrawnRect` to a local parameter? For example,

```Java
Rect drawnRect = shadowCanvas.getDrawnRect(0);
assertThat(drawnRect.left).isEqualsTo(2f);
assertThat(drawnRect.top).isEquals(4f);
...
```